### ISSUE TITLE
Documentation comments for macros are not preserved

### ISSUE THREAD

**Issue by daniel-grumberg** [llvm/llvm-project#83819]
Documentation comments for C preprocessor macros are not preserved.

**Comment by shafik** [comment]
I am not sure I understand the bug, can you provide a minimal reproducer, preferably with a godbolt link.

**Comment by daniel-grumberg** [comment]
Maybe I filtered this to the wrong place, but effectively the problem is that there is no way that I am aware of to track documentation comments attached to a macro definition.
```c
/// My fancy doc comment.
#define ADD(a, b) ((a) + (b))
```
The functionality of tracking the documentation comments associated with an AST node in the frontend already exists but there is no similar functionality with macros and their associated documentation comment.

**Comment by shafik** [comment]
CC @AaronBallman

**Comment by AaronBallman** [comment]
Correct, Clang's AST doesn't have a representation for macros so there's nothing to attach the documentation comments *to*. But we could perhaps find a way to squirrel those away with the macro definitions in the preprocessor: https://github.com/llvm/llvm-project/blob/f8fab2126ffab713f4ab4619360b6941be6d4e35/clang/include/clang/Lex/MacroInfo.h#L39

**Comment by DougGregor** [comment]
This is implemented
